src/OrchardCore.Modules/OrchardCore.Media/Shortcodes/ShareShortcodeProvider.cs

```diff
@@ -0,0 +1,174 @@
+
+        private static string GetFileNameFromUrl(string url)
+        {
+            Uri uri;
+            if (!Uri.TryCreate(url, UriKind.Absolute, out uri))
+            {
+                uri = new Uri(SomeBaseUri, url);
+            }
+
+            return Path.GetFileName(uri.LocalPath);
+        }
+
+        #endregion
+
+        public ValueTask<string> EvaluateAsync(string identifier, Arguments arguments, string content, Context context)
+        {
+            // we can't process what we don't understand, and we are discarding or filtering out anything we don't understand to keep things safe
+            if (!Shortcodes.Contains(identifier))
+            {
+                return Null;
+            }
+
+            // critical that at least the content or the href/url/source is set
+            if (String.IsNullOrWhiteSpace(content) && !IsSet(arguments, ShortcodeProperty.Url))
+            {
+                return ShareShortcode;
+            }
+
+            string url;
+            string href;
+            // sort out and infer as necessary the description (content) and resource location (url and href)
+            if (IsSet(arguments, ShortcodeProperty.Url))
+            {
+                url = UrlFixer(arguments.NamedOrDefault(ShortcodeProperty.Url.Name));
+                if (String.IsNullOrWhiteSpace(content))
+                {
+                    content = GetFileNameFromUrl(url).Split(FileExtensionSeperator)[0];
+                }
+            }
+            else
+            {
+                url = UrlFixer(content);
+                // because both the content and the href can't be empty, we presume the content was the href in this branch, and replace a shortened, friendlier name for the resulting content
+                content = GetFileNameFromUrl(content).Split(FileExtensionSeperator)[0];
+            }
+
+            // urlencode to fix space in path rendering
+            href = FormatAsHtmlElementAttribute(ShortcodeProperty.Url, Uri.EscapeUriString(url));
+            var saveAs = IsSet(arguments, ShortcodeProperty.Save) ? arguments.NamedOrDefault(ShortcodeProperty.Save.Name) : GetFileNameFromUrl(url);
+            saveAs = FormatAsHtmlElementAttribute(ShortcodeProperty.Save, saveAs);
+
+            // process contextually benign adornments
+            var classList = String.Empty;
+            var titleText = String.Empty;
+            if (arguments.Any())
+            {
+
+                classList = FormatArguement(arguments, ShortcodeProperty.Class);
+                titleText = FormatArguement(arguments, ShortcodeProperty.Tooltip);
+                //var queryStringParams = new Dictionary<string, string>();
+                //content = QueryHelpers.AddQueryString(content, queryStringParams);
+            }
+
+            content = String.Format("<a {0} {1} {2} {3}>{4}</a>", new string[] { titleText, classList, href, saveAs, content });
```

Comment:
The actual rendering doesn't make me feel good in the unnecessary rigidity/explicitness, but that is how it was essentially handled in the ImageShortcodeProvider.  It relates back to the mapping of short-code properties to Html element attributes.  The only mapping that is critical is the `content` field between the opening and closing anchor tags.  The rest of the properties could be just an enumerable splatted into the opening tag.

Reply:
You don't need a new array, just pass the arguments one after the other